# Imports and Reads

[Attribute File](data\Attributes_2015-16.csv)  
[Dataset File](\data\Data_2015-16.csv)

Steps:  
Access Data 2015-16.csv  
Columns numbers are the attribute_id.  
Pick attribute id: 315   or.   542  
Identify attributes that are correlated/dependent/related/influencing to the picked attribute. (you can pick up any approach) Use Data2015-16.csv for the same.  
Say you got : 315--- related to ( 23,67,89,54,2234)  
Join the above results with Attribute_2015-16.csv  
Store results as output.csv  

In [116]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np
import math

attr = pd.read_csv('D:\\Kartik\\IIITB\\Study\\Sem 3\\Placement Prep\\ML Projects\\PE ADT SWL\\data\\Attributes_2015-16.csv', delimiter=';')
df = pd.read_csv('D:\\Kartik\\IIITB\\Study\\Sem 3\\Placement Prep\\ML Projects\\PE ADT SWL\\data\\Data_2015-16.csv', delimiter=';')

pd.reset_option('display.max_columns')
pd.reset_option('display.max_rows')

# reviewing dataset

In [117]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30 entries, 0 to 29
Columns: 1169 entries, 1 to 1183
dtypes: float64(430), int64(738), object(1)
memory usage: 274.1+ KB


In [118]:
dtypedf = pd.DataFrame(df.dtypes)
int64df = dtypedf[dtypedf[0]=='int64']
int64df['index'] = int64df.index.to_list()
int64df

C:\Users\karti\AppData\Local\Temp\ipykernel_24368\3721613660.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  int64df['index'] = int64df.index.to_list()


0 index
1     int64     1
3     int64     3
4     int64     4
5     int64     5
6     int64     6
...     ...   ...
1179  int64  1179
1180  int64  1180
1181  int64  1181
1182  int64  1182
1183  int64  1183

[738 rows x 2 columns]

let's convert all dtypes to float64 for consistency between data

In [119]:
# attr = pd.read_csv('D:\\Kartik\\IIITB\\Study\\Sem 3\\Placement Prep\\ML Projects\\PE ADT SWL\\data\\Attributes_2015-16.csv', delimiter=';'
df = pd.read_csv('D:\\Kartik\\IIITB\\Study\\Sem 3\\Placement Prep\\ML Projects\\PE ADT SWL\\data\\Data_2015-16.csv', delimiter=';'\
                   , dtype={col: 'float64' for col in int64df['index']})

In [120]:
dtypedf = pd.DataFrame(df.dtypes)
int64df = dtypedf[dtypedf[0]=='int64']
int64df['index'] = int64df.index.to_list()
int64df

Empty DataFrame
Columns: [0, index]
Index: []

In [121]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30 entries, 0 to 29
Columns: 1169 entries, 1 to 1183
dtypes: float64(1168), object(1)
memory usage: 274.1+ KB


In [122]:
df

1                2     3      4     5      6     7     8       9  \
0    1.0        Bengaluru  27.0  198.0  27.0   96.0   4.0  16.0   562.0   
1    2.0    Bengaluru (R)  17.0  183.0  17.0  105.0   4.0   7.0   957.0   
2    3.0       Ramanagara  18.0  244.0  18.0  126.0   4.0   6.0   820.0   
3    4.0      Chitradurga  22.0  312.0  22.0  189.0   6.0   8.0   948.0   
4    5.0        Davangere  24.0  272.0  24.0  233.0   6.0   8.0   800.0   
5    6.0            Kolar  27.0  262.0  27.0  156.0   5.0   7.0  1608.0   
6    7.0  Chikkaballapura  26.0  249.0  26.0  157.0   6.0   8.0  1324.0   
7    8.0       Shivamogga  37.0  377.0  37.0  271.0   7.0   9.0  1444.0   
8    9.0         Tumakuru  50.0  568.0  50.0  331.0  10.0  12.0  2582.0   
9   10.0   Chikkamagaluru  34.0  237.0  34.0  227.0   7.0   9.0  1022.0   
10  11.0       D. Kannada  17.0  289.0  17.0  230.0   5.0  41.0   331.0   
11  12.0            Udupi   9.0  187.0   9.0  158.0   3.0  20.0   233.0   
12  13.0           Hassan  38.0  411.0  38.0  267.0   8.0  11.0  2418.0   
13  14.0           Kodagu  16.0  109.0  16.0  104.0   3.0   5.0   291.0   
14  15.0           Mandya  45.0  437.0  45.0  234.0   7.0   9.0  1368.0   
15  16.0           Mysuru  33.0  448.0  33.0  266.0   7.0  19.0  1199.0   
16  17.0    Chamarajnagar  16.0  231.0  16.0  130.0   4.0   5.0   428.0   
17  18.0         Belagavi  35.0  554.0  35.0  502.0  10.0  31.0  1263.0   
18  19.0       Vijayapura  20.0  240.0  20.0  213.0   5.0   6.0   679.0   
19  20.0        Bagalkote  18.0  251.0  18.0  198.0   6.0  15.0   613.0   
20  21.0          Dharwad  14.0  202.0  14.0  144.0   5.0   6.0   361.0   
21  22.0            Gadag  11.0  197.0  11.0  122.0   5.0   9.0   322.0   
22  23.0           Haveri  19.0  285.0  19.0  223.0   7.0  10.0   696.0   
23  24.0   Uttara Kannada  35.0  243.0  35.0  231.0  11.0  18.0  1243.0   
24  25.0          Ballari  27.0  295.0  27.0  199.0   7.0  13.0   522.0   
25  26.0            Bidar  29.0  330.0  29.0  186.0   5.0   7.0   595.0   
26  27.0       Kalaburagi  32.0  350.0  32.0  264.0   7.0  11.0   871.0   
27  28.0         Yadagiri  16.0  172.0  16.0  123.0   3.0   7.0   487.0   
28  29.0          Raichur  37.0  280.0  37.0  181.0   5.0   9.0   815.0   
29  30.0           Koppal  20.0  161.0  20.0  153.0   4.0   5.0   595.0   

       10  ...  1174      1175      1176      1177    1178      1179  \
0    26.0  ...   0.0   73454.0   65846.0   14679.0    90.0   80615.0   
1    94.0  ...   0.0  133665.0  145020.0   33789.0    58.0  178867.0   
2    52.0  ...   0.0  183810.0  221460.0   49825.0   163.0  271448.0   
3   115.0  ...   0.0  402224.0  227197.0   63232.0   148.0  290577.0   
4   123.0  ...   0.0  323462.0  226426.0   54817.0  1434.0  282677.0   
5   189.0  ...   0.0  189103.0  198213.0   38257.0   385.0  236855.0   
6   189.0  ...   0.0  170903.0  178301.0   36054.0   107.0  214462.0   
7    86.0  ...   0.0  234739.0  169524.0   44187.0   297.0  214008.0   
8   133.0  ...   0.0  569246.0  326779.0   87403.0   246.0  414428.0   
9    95.0  ...   0.0  273004.0  182860.0   39251.0  1366.0  223477.0   
10    NaN  ...   0.0  164504.0  142792.0   65458.0   847.0  209097.0   
11    NaN  ...   0.0  140138.0  112972.0   86970.0   543.0  200485.0   
12  156.0  ...   0.0  408846.0  341883.0   93219.0   579.0  435681.0   
13    5.0  ...   0.0  154512.0   57651.0   10803.0   427.0   68881.0   
14  109.0  ...   0.0  283994.0  280482.0  112940.0  2344.0  395766.0   
15  137.0  ...   0.0  287892.0  301928.0   83606.0   192.0  385726.0   
16   81.0  ...   0.0  170713.0  172297.0   39453.0   446.0  212196.0   
17   12.0  ...   0.0  915122.0  487883.0   42948.0   104.0  530935.0   
18   13.0  ...   0.0  844500.0  300565.0   41600.0    38.0  342203.0   
19   11.0  ...   0.0  439317.0  193388.0   34298.0    45.0  227731.0   
20   18.0  ...   0.0  324861.0  124153.0   18868.0   204.0  143225.0   
21   15.0  ...   0.0  353433.0  141852.0   21297.0   317.0  163466.0   
22    7.0  ...   0.0  329

now all dtypes are float64

In [123]:
df.head()

1              2     3      4     5      6    7     8      9     10  ...  \
0  1.0      Bengaluru  27.0  198.0  27.0   96.0  4.0  16.0  562.0   26.0  ...   
1  2.0  Bengaluru (R)  17.0  183.0  17.0  105.0  4.0   7.0  957.0   94.0  ...   
2  3.0     Ramanagara  18.0  244.0  18.0  126.0  4.0   6.0  820.0   52.0  ...   
3  4.0    Chitradurga  22.0  312.0  22.0  189.0  6.0   8.0  948.0  115.0  ...   
4  5.0      Davangere  24.0  272.0  24.0  233.0  6.0   8.0  800.0  123.0  ...   

   1174      1175      1176     1177    1178      1179      1180      1181  \
0   0.0   73454.0   65846.0  14679.0    90.0   80615.0   65336.0   13465.0   
1   0.0  133665.0  145020.0  33789.0    58.0  178867.0  125693.0   26260.0   
2   0.0  183810.0  221460.0  49825.0   163.0  271448.0  167899.0   34406.0   
3   0.0  402224.0  227197.0  63232.0   148.0  290577.0  438437.0  117797.0   
4   0.0  323462.0  226426.0  54817.0  1434.0  282677.0  333819.0   75816.0   

     1182      1183  
0   544.0   79345.0  
1    81.0  152034.0  
2   261.0  202566.0  
3  4543.0  560777.0  
4  4051.0  413686.0  

[5 rows x 1169 columns]

In [124]:
df.columns

Index(['1', '2', '3', '4', '5', '6', '7', '8', '9', '10',
       ...
       '1174', '1175', '1176', '1177', '1178', '1179', '1180', '1181', '1182',
       '1183'],
      dtype='object', length=1169)

In [125]:
# pd.set_option('display.max_columns')
# pd.set_option('display.max_rows',25)
df

1                2     3      4     5      6     7     8       9  \
0    1.0        Bengaluru  27.0  198.0  27.0   96.0   4.0  16.0   562.0   
1    2.0    Bengaluru (R)  17.0  183.0  17.0  105.0   4.0   7.0   957.0   
2    3.0       Ramanagara  18.0  244.0  18.0  126.0   4.0   6.0   820.0   
3    4.0      Chitradurga  22.0  312.0  22.0  189.0   6.0   8.0   948.0   
4    5.0        Davangere  24.0  272.0  24.0  233.0   6.0   8.0   800.0   
5    6.0            Kolar  27.0  262.0  27.0  156.0   5.0   7.0  1608.0   
6    7.0  Chikkaballapura  26.0  249.0  26.0  157.0   6.0   8.0  1324.0   
7    8.0       Shivamogga  37.0  377.0  37.0  271.0   7.0   9.0  1444.0   
8    9.0         Tumakuru  50.0  568.0  50.0  331.0  10.0  12.0  2582.0   
9   10.0   Chikkamagaluru  34.0  237.0  34.0  227.0   7.0   9.0  1022.0   
10  11.0       D. Kannada  17.0  289.0  17.0  230.0   5.0  41.0   331.0   
11  12.0            Udupi   9.0  187.0   9.0  158.0   3.0  20.0   233.0   
12  13.0           Hassan  38.0  411.0  38.0  267.0   8.0  11.0  2418.0   
13  14.0           Kodagu  16.0  109.0  16.0  104.0   3.0   5.0   291.0   
14  15.0           Mandya  45.0  437.0  45.0  234.0   7.0   9.0  1368.0   
15  16.0           Mysuru  33.0  448.0  33.0  266.0   7.0  19.0  1199.0   
16  17.0    Chamarajnagar  16.0  231.0  16.0  130.0   4.0   5.0   428.0   
17  18.0         Belagavi  35.0  554.0  35.0  502.0  10.0  31.0  1263.0   
18  19.0       Vijayapura  20.0  240.0  20.0  213.0   5.0   6.0   679.0   
19  20.0        Bagalkote  18.0  251.0  18.0  198.0   6.0  15.0   613.0   
20  21.0          Dharwad  14.0  202.0  14.0  144.0   5.0   6.0   361.0   
21  22.0            Gadag  11.0  197.0  11.0  122.0   5.0   9.0   322.0   
22  23.0           Haveri  19.0  285.0  19.0  223.0   7.0  10.0   696.0   
23  24.0   Uttara Kannada  35.0  243.0  35.0  231.0  11.0  18.0  1243.0   
24  25.0          Ballari  27.0  295.0  27.0  199.0   7.0  13.0   522.0   
25  26.0            Bidar  29.0  330.0  29.0  186.0   5.0   7.0   595.0   
26  27.0       Kalaburagi  32.0  350.0  32.0  264.0   7.0  11.0   871.0   
27  28.0         Yadagiri  16.0  172.0  16.0  123.0   3.0   7.0   487.0   
28  29.0          Raichur  37.0  280.0  37.0  181.0   5.0   9.0   815.0   
29  30.0           Koppal  20.0  161.0  20.0  153.0   4.0   5.0   595.0   

       10  ...  1174      1175      1176      1177    1178      1179  \
0    26.0  ...   0.0   73454.0   65846.0   14679.0    90.0   80615.0   
1    94.0  ...   0.0  133665.0  145020.0   33789.0    58.0  178867.0   
2    52.0  ...   0.0  183810.0  221460.0   49825.0   163.0  271448.0   
3   115.0  ...   0.0  402224.0  227197.0   63232.0   148.0  290577.0   
4   123.0  ...   0.0  323462.0  226426.0   54817.0  1434.0  282677.0   
5   189.0  ...   0.0  189103.0  198213.0   38257.0   385.0  236855.0   
6   189.0  ...   0.0  170903.0  178301.0   36054.0   107.0  214462.0   
7    86.0  ...   0.0  234739.0  169524.0   44187.0   297.0  214008.0   
8   133.0  ...   0.0  569246.0  326779.0   87403.0   246.0  414428.0   
9    95.0  ...   0.0  273004.0  182860.0   39251.0  1366.0  223477.0   
10    NaN  ...   0.0  164504.0  142792.0   65458.0   847.0  209097.0   
11    NaN  ...   0.0  140138.0  112972.0   86970.0   543.0  200485.0   
12  156.0  ...   0.0  408846.0  341883.0   93219.0   579.0  435681.0   
13    5.0  ...   0.0  154512.0   57651.0   10803.0   427.0   68881.0   
14  109.0  ...   0.0  283994.0  280482.0  112940.0  2344.0  395766.0   
15  137.0  ...   0.0  287892.0  301928.0   83606.0   192.0  385726.0   
16   81.0  ...   0.0  170713.0  172297.0   39453.0   446.0  212196.0   
17   12.0  ...   0.0  915122.0  487883.0   42948.0   104.0  530935.0   
18   13.0  ...   0.0  844500.0  300565.0   41600.0    38.0  342203.0   
19   11.0  ...   0.0  439317.0  193388.0   34298.0    45.0  227731.0   
20   18.0  ...   0.0  324861.0  124153.0   18868.0   204.0  143225.0   
21   15.0  ...   0.0  353433.0  141852.0   21297.0   317.0  163466.0   
22    7.0  ...   0.0  329

In [126]:
dfnotna = pd.DataFrame(df.notna().sum())
completeNaDf = dfnotna[dfnotna[0]==0] #completely null attributes
completeNaDf.shape

(14, 1)

In [127]:
completeNaDf

0
419  0
507  0
508  0
509  0
590  0
600  0
676  0
691  0
692  0
693  0
694  0
766  0
863  0
934  0

there are 14 completely null columns let's remove them

In [128]:
df['419']

0    NaN
1    NaN
2    NaN
3    NaN
4    NaN
5    NaN
6    NaN
7    NaN
8    NaN
9    NaN
10   NaN
11   NaN
12   NaN
13   NaN
14   NaN
15   NaN
16   NaN
17   NaN
18   NaN
19   NaN
20   NaN
21   NaN
22   NaN
23   NaN
24   NaN
25   NaN
26   NaN
27   NaN
28   NaN
29   NaN
Name: 419, dtype: float64

In [129]:
df = df.drop(completeNaDf.index.to_list(), axis=1)
df.shape

(30, 1155)

In [130]:
# df['419']

now there are only semi null attributes but no completely null attributes

In [131]:
nadf = pd.DataFrame(df.isna().sum())
nadf = nadf[nadf[0]>0]
nadf['index'] = nadf.index.to_list()
nadf

0 index
10    2    10
12   20    12
13    2    13
14    1    14
15    6    15
..   ..   ...
929   1   929
939   2   939
941  26   941
942  21   942
943   7   943

[327 rows x 2 columns]

In [132]:
nadf.shape

(327, 2)

we have 327 na present attributes. since its all numeric data. let's replace by mean.

In [133]:
df.describe()

1          3           4          5           6          7  \
count  30.000000  30.000000   30.000000  30.000000   30.000000  30.000000   
mean   15.500000  25.633333  285.800000  25.633333  200.633333   5.866667   
std     8.803408  10.223311  109.660102  10.223311   81.977913   2.046584   
min     1.000000   9.000000  109.000000   9.000000   96.000000   3.000000   
25%     8.250000  17.250000  209.250000  17.250000  146.250000   4.250000   
50%    15.500000  25.000000  256.500000  25.000000  193.500000   5.500000   
75%    22.750000  33.750000  325.500000  33.750000  232.500000   7.000000   
max    30.000000  50.000000  568.000000  50.000000  502.000000  11.000000   

               8            9          10           11  ...  1174  \
count  30.000000    30.000000   28.000000    30.000000  ...  30.0   
mean   11.566667   913.233333   69.392857   978.000000  ...   0.0   
std     7.933705   571.791177   56.548464   614.694934  ...   0.0   
min     5.000000   233.000000    5.000000   233.000000  ...   0.0   
25%     7.000000   532.000000   23.250000   561.000000  ...   0.0   
50%     9.000000   807.500000   49.500000   878.000000  ...   0.0   
75%    12.750000  1232.000000  110.500000  1285.500000  ...   0.0   
max    41.000000  2582.000000  189.000000  2715.000000  ...   0.0   

                1175           1176           1177         1178  \
count      30.000000      30.000000      30.000000    30.000000   
mean   343638.266667  211102.800000   49549.300000   420.866667   
std    210942.131610   89507.381766   24881.744584   515.071855   
min     73454.000000   57651.000000   10803.000000    11.000000   
25%    174129.750000  154623.000000   34737.000000    93.500000   
50%    324161.500000  195720.000000   42803.000000   258.000000   
75%    406577.000000  263341.750000   63063.250000   492.500000   
max    915122.000000  487883.000000  112940.000000  2344.000000   

                1179           1180           1181          1182  \
count      30.000000      30.000000      30.000000     30.000000   
mean   261072.966667  336340.233333   63277.466667   2430.866667   
std    104618.868770  218242.860293   33861.683741   2908.921897   
min     68881.000000   65336.000000   13465.000000     81.000000   
25%    209871.750000  180286.000000   34986.000000    786.500000   
50%    227094.500000  299976.000000   61853.000000   1537.000000   
75%    321669.000000  409315.250000   76305.000000   3122.500000   
max    530935.000000  919337.000000  157393.000000  13609.000000   

                1183  
count      30.000000  
mean   402048.566667  
std    244156.324400  
min     79345.000000  
25%    217946.500000  
50%    366015.000000  
75%    478280.750000  
max    992289.000000  

[8 rows x 1154 columns]

In [134]:
for index in nadf['index']:
    df[index] = df[index].fillna(df[index].mean())

In [135]:
nadf = pd.DataFrame(df.isna().sum())
nadf = nadf[nadf[0]>0]
nadf['index'] = nadf.index.to_list()
nadf

Empty DataFrame
Columns: [0, index]
Index: []

now there are no null values at all in dataset

In [137]:
df

1                2     3      4     5      6     7     8       9  \
0    1.0        Bengaluru  27.0  198.0  27.0   96.0   4.0  16.0   562.0   
1    2.0    Bengaluru (R)  17.0  183.0  17.0  105.0   4.0   7.0   957.0   
2    3.0       Ramanagara  18.0  244.0  18.0  126.0   4.0   6.0   820.0   
3    4.0      Chitradurga  22.0  312.0  22.0  189.0   6.0   8.0   948.0   
4    5.0        Davangere  24.0  272.0  24.0  233.0   6.0   8.0   800.0   
5    6.0            Kolar  27.0  262.0  27.0  156.0   5.0   7.0  1608.0   
6    7.0  Chikkaballapura  26.0  249.0  26.0  157.0   6.0   8.0  1324.0   
7    8.0       Shivamogga  37.0  377.0  37.0  271.0   7.0   9.0  1444.0   
8    9.0         Tumakuru  50.0  568.0  50.0  331.0  10.0  12.0  2582.0   
9   10.0   Chikkamagaluru  34.0  237.0  34.0  227.0   7.0   9.0  1022.0   
10  11.0       D. Kannada  17.0  289.0  17.0  230.0   5.0  41.0   331.0   
11  12.0            Udupi   9.0  187.0   9.0  158.0   3.0  20.0   233.0   
12  13.0           Hassan  38.0  411.0  38.0  267.0   8.0  11.0  2418.0   
13  14.0           Kodagu  16.0  109.0  16.0  104.0   3.0   5.0   291.0   
14  15.0           Mandya  45.0  437.0  45.0  234.0   7.0   9.0  1368.0   
15  16.0           Mysuru  33.0  448.0  33.0  266.0   7.0  19.0  1199.0   
16  17.0    Chamarajnagar  16.0  231.0  16.0  130.0   4.0   5.0   428.0   
17  18.0         Belagavi  35.0  554.0  35.0  502.0  10.0  31.0  1263.0   
18  19.0       Vijayapura  20.0  240.0  20.0  213.0   5.0   6.0   679.0   
19  20.0        Bagalkote  18.0  251.0  18.0  198.0   6.0  15.0   613.0   
20  21.0          Dharwad  14.0  202.0  14.0  144.0   5.0   6.0   361.0   
21  22.0            Gadag  11.0  197.0  11.0  122.0   5.0   9.0   322.0   
22  23.0           Haveri  19.0  285.0  19.0  223.0   7.0  10.0   696.0   
23  24.0   Uttara Kannada  35.0  243.0  35.0  231.0  11.0  18.0  1243.0   
24  25.0          Ballari  27.0  295.0  27.0  199.0   7.0  13.0   522.0   
25  26.0            Bidar  29.0  330.0  29.0  186.0   5.0   7.0   595.0   
26  27.0       Kalaburagi  32.0  350.0  32.0  264.0   7.0  11.0   871.0   
27  28.0         Yadagiri  16.0  172.0  16.0  123.0   3.0   7.0   487.0   
28  29.0          Raichur  37.0  280.0  37.0  181.0   5.0   9.0   815.0   
29  30.0           Koppal  20.0  161.0  20.0  153.0   4.0   5.0   595.0   

            10  ...  1174      1175      1176      1177    1178      1179  \
0    26.000000  ...   0.0   73454.0   65846.0   14679.0    90.0   80615.0   
1    94.000000  ...   0.0  133665.0  145020.0   33789.0    58.0  178867.0   
2    52.000000  ...   0.0  183810.0  221460.0   49825.0   163.0  271448.0   
3   115.000000  ...   0.0  402224.0  227197.0   63232.0   148.0  290577.0   
4   123.000000  ...   0.0  323462.0  226426.0   54817.0  1434.0  282677.0   
5   189.000000  ...   0.0  189103.0  198213.0   38257.0   385.0  236855.0   
6   189.000000  ...   0.0  170903.0  178301.0   36054.0   107.0  214462.0   
7    86.000000  ...   0.0  234739.0  169524.0   44187.0   297.0  214008.0   
8   133.000000  ...   0.0  569246.0  326779.0   87403.0   246.0  414428.0   
9    95.000000  ...   0.0  273004.0  182860.0   39251.0  1366.0  223477.0   
10   69.392857  ...   0.0  164504.0  142792.0   65458.0   847.0  209097.0   
11   69.392857  ...   0.0  140138.0  112972.0   86970.0   543.0  200485.0   
12  156.000000  ...   0.0  408846.0  341883.0   93219.0   579.0  435681.0   
13    5.000000  ...   0.0  154512.0   57651.0   10803.0   427.0   68881.0   
14  109.000000  ...   0.0  283994.0  280482.0  112940.0  2344.0  395766.0   
15  137.000000  ...   0.0  287892.0  301928.0   83606.0   192.0  385726.0   
16   81.000000  ...   0.0  170713.0  172297.0   39453.0   446.0  212196.0   
17   12.000000  ...   0.0  915122.0  487883.0   42948.0   104.0  530935.0   
18   13.000000  ...   0.0  844500.0  300565.0   41600.0    38.0  342203.0   
19   11.000000  ...   0.0  439317.0  193388.0   34298.0    45.0  227731.0   
20   18.000000  ...   0.0  324861.0  124153.0   18868.0   204.0 

In [138]:
df.shape

(30, 1155)

now there are 1155 attributes remaining including one object attribute

In [141]:
numericAttr = [ attr for attr in df.columns if attr!='2']

In [143]:
df[numericAttr].corr()

1         3         4         5         6         7         8  \
1     1.000000 -0.031226 -0.055436 -0.031226  0.083019  0.072729 -0.046162   
3    -0.031226  1.000000  0.775379  1.000000  0.627085  0.735927  0.078325   
4    -0.055436  0.775379  1.000000  0.775379  0.860198  0.746601  0.354629   
5    -0.031226  1.000000  0.775379  1.000000  0.627085  0.735927  0.078325   
6     0.083019  0.627085  0.860198  0.627085  1.000000  0.801677  0.511643   
...        ...       ...       ...       ...       ...       ...       ...   
1179  0.132552  0.629821  0.844278  0.629821  0.796747  0.592889  0.223547   
1180  0.444796  0.274452  0.465804  0.274452  0.598305  0.356064  0.045748   
1181  0.323749  0.349637  0.457764  0.349637  0.422103  0.274567  0.015522   
1182 -0.155760  0.112613 -0.051848  0.112613  0.104988  0.079193 -0.049705   
1183  0.440632  0.295155  0.479235  0.295155  0.594595  0.357297  0.042453   

             9        10        11  ...  1174      1175      1176      1177  \
1    -0.298104 -0.542614 -0.321925  ...   NaN  0.422923  0.171396 -0.056077   
3     0.793698  0.413808  0.784653  ...   NaN  0.296818  0.585241  0.536886   
4     0.708748  0.328380  0.691823  ...   NaN  0.504548  0.810805  0.629888   
5     0.793698  0.413808  0.784653  ...   NaN  0.296818  0.585241  0.536886   
6     0.532115  0.098359  0.504336  ...   NaN  0.643170  0.812483  0.424450   
...        ...       ...       ...  ...   ...       ...       ...       ...   
1179  0.594378  0.276379  0.581614  ...   NaN  0.724303  0.978490  0.682012   
1180  0.136118 -0.257042  0.112143  ...   NaN  0.987257  0.773616  0.121351   
1181  0.202541  0.035859  0.192463  ...   NaN  0.744467  0.604316  0.538957   
1182 -0.010706  0.059235 -0.003732  ...   NaN -0.052963 -0.066459 -0.002506   
1183  0.149633 -0.224082  0.126889  ...   NaN  0.985093  0.774528  0.183189   

          1178      1179      1180      1181      1182      1183  
1    -0.152277  0.132552  0.444796  0.323749 -0.155760  0.440632  
3     0.289664  0.629821  0.274452  0.349637  0.112613  0.295155  
4     0.158656  0.844278  0.465804  0.457764 -0.051848  0.479235  
5     0.289664  0.629821  0.274452  0.349637  0.112613  0.295155  
6     0.136938  0.796747  0.598305  0.422103  0.104988  0.594595  
...        ...       ...       ...       ...       ...       ...  
1179  0.130355  1.000000  0.689580  0.645056 -0.055029  0.705198  
1180 -0.234210  0.689580  1.000000  0.739824 -0.081009  0.995505  
1181 -0.030759  0.645056  0.739824  1.000000 -0.023102  0.799716  
1182  0.492859 -0.055029 -0.081009 -0.023102  1.000000 -0.063701  
1183 -0.207746  0.705198  0.995505  0.799716 -0.063701  1.000000  

[1154 rows x 1154 columns]

there are na values for this cause corr is not defined for a sample with variance zero since we use variance in denominator for find the correlation

In [148]:
dfR2 = df[numericAttr].corr().pow(2) #R2 for the attributes.

In [152]:
dfR2 = dfR2.applymap(lambda x: 0 if x<0.9 else x)

C:\Users\karti\AppData\Local\Temp\ipykernel_24368\3124491555.py:1: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.
  dfR2 = dfR2.applymap(lambda x: 0 if x<0.9 else x)


In [153]:
dfR2[dfR2['315']>0]['315']

312    0.978341
313    0.936766
315    1.000000
Name: 315, dtype: float64

so attributes 312 and 313 are explaining there 90% variance from the relationship with 315